### PR TITLE
Added headphones support

### PR DESCRIPTION
**Pull request by chakrihacker**
## Description

Fixed issue #643 

Added `isHeadphonesConnected()` that allows checking wired headset or Bluetooth headphones connected status

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |
| Windows |    ❌     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [x] I added the documentation in `README.md`
* [ ] I mentioned this change in `CHANGELOG.md`
* [x] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [x] I updated the dummy web/test polyfill (`default/index.js`)
* [x] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

**Comment by mikehardy** [comment]
Oh this is very cool! Thank you!
Let me get the current release out (working on that this afternoon) then I'll review this and test it and hopefully get it out as a minor version after if it looks good

File: src/index.ts [hunk 1]
```diff
@@ -1141,6 +1141,14 @@ export function isLocationEnabledSync() {
   return false;
 }
 
+export function isHeadphonesConnected() {
+  return RNDeviceInfo.isHeadphonesConnected();
```
Comment (mikehardy) [inline src/index.ts]: These need to be restricted to android and ios
```suggestion
  if (OS === 'android' || OS === 'ios') {
    return RNDeviceInfo.isHeadphonesConnected();
  }
  return false;
```

File: src/index.ts [hunk 2]
```diff
@@ -1141,6 +1141,14 @@ export function isLocationEnabledSync() {
...
+}
+
+export function isHeadphonesConnectedSync() {
+  return RNDeviceInfo.isHeadphonesConnectedSync();
```
Comment (mikehardy) [inline src/index.ts]:
```suggestion
  if (OS === 'android' || OS === 'ios') {
    return RNDeviceInfo.isHeadphonesConnectedSync();
  }
  return false;
```

**Comment by mikehardy** [comment]
oh one other thing - if you could add it to example/App.js and README (the method table, and the example section) - that helps show usage

**Comment by chakrihacker** [comment]
i will finish it today

**Comment by chakrihacker** [comment]
Where should I add polyfill for web??

**Comment by mikehardy** [comment]
Oh good question - I restructured the code so the web is no longer a polyfill, it stays up to date automatically. I need to update the PR template to remove that bit - ignore web polyfill :-)

**Comment by chakrihacker** [comment]
I have tested Bluetooth and normal headphones for android, and for ios Bluetooth is tested. Need someone to test wired headset of type-c

**Comment by mikehardy** [comment]
Cool - do you think this is ready to go from your perspective then? I won't have all the headphones types to check either but for a new feature if there is a problem someone else can try other types and fix it - it's impossible to break current users because there aren't any and you can only test what you have access to...

**Comment by chakrihacker** [comment]
It's ready to go then

**Comment by mikehardy** [comment]
Sorry missed this in my last sweep but it does look good to go, thank! Will release a new minor version in a little bit with this included

**Pull request merged**